### PR TITLE
[Hub Generated] Review request for Microsoft.DocumentDB to add version stable/2021-01-15

### LINKED ISSUE #11222: CosmosDB Compound Index Sort Order returns incorrect case for `CompositePathSortOrder` in `Get`
API Version: 2020-04-01

Line Reference: https://github.com/Azure/azure-rest-api-specs/blob/d43b10e044e9e12e1f53a3cc2a9d3d9f01b08d90/specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2020-04-01/cosmos-db.json#L7062-L7069

The API call returns `descending` or `ascending`, which does not match the Enum values of `Descending` and `Ascending` respectively, resulting in the need to add additional comparison / conversion code that should not be necessary.

### PR DESCRIPTION
**Pull request by kristynhamasaki**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/kristynhamasaki/azure-rest-api-specs/kristynh-0115...master/).

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [X] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
Changes will be in Prod hopefully by end of January.

  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
End of January

  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
N/A
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)
Please give us an exception for linting failure "RequiredSystemDataInNewApiVersions" We are committed to incorporate this change in Cobalt and many changes in this PR are time critical.

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [X] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by kristynhamasaki** [comment]
We are seeing linting failures with RequiredSystemDataInNewApiVersions but are asking for an exception to this for now. Many of the changes here are time critical and we are committed to incorporate this change in Cobalt.

**Comment by kristynhamasaki** [comment]
Hi @lirenhe @zhenglaizhang could you please take a look? thanks!

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-01-15/cosmos-db.json [hunk 1]
```diff
@@ -0,0 +1,8571 @@
...
+      "description": "Parameters to create and update Cosmos DB Cassandra table.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties to create and update Azure Cosmos DB Cassandra table.",
+          "$ref": "#/definitions/CassandraTableCreateUpdateProperties"
+        }
+      },
+      "al
```
Comment (MehaKaushik) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-01-15/cosmos-db.json:8329]: Please add "4.0" as well. Thanks.

Comment (kristynhamasaki) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-01-15/cosmos-db.json]: done

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-01-15/cosmos-db.json [hunk 2]
```diff
@@ -0,0 +1,8572 @@
...
+          "description": "The type of Azure resource."
+        },
+        "location": {
+          "type": "string",
+          "description": "The location of the resource group to which the resource belongs."
+        },
+        "tags": {
+          "$ref": "#/definitions/Tags"
+        },
+        "identity": {
```
Comment (pilor) [inline specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-01-15/cosmos-db.json:5348]: All of the resources utilizing this top level definition actualyl support managed identity?

**Review by pilor — COMMENTED** [review]
What steps are you taking to ensure you don't have a time crunch that prevents you from adding systemData next time you need to update an api-version?

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-01-15/cosmos-db.json (hunk 2 above)
Comment (MehaKaushik) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-01-15/cosmos-db.json:5348]: I believe it is only supported by our top level resource, DatabaseAccount. It should not be added in other places. @xujin-zhang please fix it.

**Comment by MehaKaushik** [comment]
> What steps are you taking to ensure you don't have a time crunch that prevents you from adding systemData next time you need to update an api-version?

Honestly, we were not aware of any timelines for this change to be made, hence we did not prioritize this earlier. Given the fact that this time we were delayed in releasing our swagger specs, we are prioritizing this fix before the next release. It is tracked internally as work-item 787607.

File: specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-01-15/cosmos-db.json (hunk 2 above)
Comment (kristynhamasaki) [reply specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2021-01-15/cosmos-db.json]: Fix has been made!

**Comment by kristynhamasaki** [comment]
Hi @pilor we have made the changes for identity. Please take a look when you can, thanks!

**Comment by pilor** [comment]
> Hi @pilor we have made the changes for identity. Please take a look when you can, thanks!

The "identity" property was moved inside the property bag of the resource which is not correct.  It should be a top level property.  The problem before was that it was in the base definition which I think was used by more than just the resources that support identity.

**Comment by xujin-zhang** [comment]
> > Hi @pilor we have made the changes for identity. Please take a look when you can, thanks!
> 
> The "identity" property was moved inside the property bag of the resource which is not correct. It should be a top level property. The problem before was that it was in the base definition which I think was used by more than just the resources that support identity.

@pilor Hi, I have made changes according to your needs!

File: specification/cosmos-db/resource-manager/readme.md [hunk 3]
```diff
@@ -27,21 +27,33 @@ These are the global settings for the Cosmos-DB API.
...
-tag: package-2020-09
+tag: package-2021-01
 ```
 
 
+### Tag: package-2021-01
+
+These settings apply only when `--tag=package-2021-01` is specified on the command line.
+
+```yaml $(tag) == 'package-2021-01'
```
Comment (MehaKaushik) [inline specification/cosmos-db/resource-manager/readme.md:38]: You should also make changes in other Readme files in this folder.

Comment (kristynhamasaki) [reply specification/cosmos-db/resource-manager/readme.md:38]: done

**Comment by MehaKaushik** [comment]
Filed the breaking change request here https://msazure.visualstudio.com/One/_workitems/edit/9236496

**Review by JeffreyRichter — APPROVED** [review]

**Review by MehaKaushik — APPROVED** [review]

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**